### DISCUSSION TITLE
popup edit with dynamic columns v-for in QTable

### DISCUSSION THREAD

**Discussion by ameet123** [quasarframework/quasar#13251] [General - Components / Directives / etc] [answered]
Hi,
I am trying to add q-popup edit and q-input as per the example in QTable. - https://quasar.dev/vue-components/table#popup-editing
However, the example describes each column hard-coded. 
I don't know the names of my column ahead of time and would like to do this dynamically.
I have it so far as,
```
 <q-table title="Treats" :rows="rows" :columns="columns" row-key="name">
        <template v-slot:body="props">
          <q-tr :props="props">
            <q-td v-for="col in props.cols" :key="col.name" :props="props">
              {{ col.value }}
              <q-popup-edit v-model="col.value" v-slot="scope" auto-save>
                <q-input v-model="scope.value" dense autofocus counter />
              </q-popup-edit>
            </q-td>
          </q-tr>
        </template>
      </q-table>
```
This allows me to click each cell and enter a new value. But it doesn't `save` it. 
What am I missing? 
thanks much

**Update**
ok, after some experimentation,
```
<q-popup-edit
                v-model="props.row[col.name]"
                v-slot="scope"
                auto-save
                buttons
              >
```
changing the v-model on q-popup-edit actually changes the data, but does not show up in the table. If I click on the table, go to next page etc. then the change shows up.

**Comment by metalsadman** [comment, marked as answer]
works fine https://codepen.io/metalsadman/pen/oNpRpwB

check if the data in `rows` prop are reactive.

**Reply by ameet123** [reply]
thanks @metalsadman making the rows array wrapped in `ref` did the trick. thanks much.
